Batteries/Data/Vector/Unexpanders.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2024 Shreyas Srinivas. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shreyas Srinivas, Francois Dorais
+-/
+
+import Batteries.Data.Vector.Basic
+
+/-!
+## Vector.Unexpanders
+
+The definitions are used to provide unexpanders in metaprogramming. They correspond to similar array definitions `mkArray0`,...,`mkArray8` and their unexpanders in
```

Comment:
```suggestion
The definitions `mkVector*` provide convenient ways to write terms for short vectors, especially in metaprogramming. Unexpanders are provided make these shortcuts invisible to users.
```

Reply:
(Please check for long lines.)

Reply:
Yeah got those. vscode was wrapping around automatically making them harder to notice.

Reply:
Done.